src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let initialState = { ...state };
+
+  actions.forEach((action) => {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      initialState = { ...initialState, ...extraData };
+    } else if (type === 'removeProperties') {
+      initialState = { ...initialState };
+
+      keysToRemove.forEach((key) => {
+        delete initialState[key];
+      });
+    } else if (type === 'clear') {
+      initialState = {};
+    } else {
+      throw new Error(`Unknown action type: ${type}`);
```

Comment:
Consider using a `switch` statement instead of multiple `if-else` chains as it is more suitable for handling different cases based on the value of `type`. It will also help in adhering to the checklist point about not repeating yourself, as the push operation can be done once after the switch.